File: Jobby.Clients/jobby-razor/Contracts/Contact/CreateContactRequest.cs
using Jobby.Client.Models;

namespace Jobby.Client.Contracts.Contact;

public class CreateContactRequest
{
    public Guid BoardId { get; set; }
    public List<Guid> JobIds { get; set; }
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public string JobTitle { get; set; }
    public string Location { get; set; }
    public Social Socials { get; set; }
    public List<Email> Emails { get; set; }
    public List<Phone> Phones { get; set; }
    public List<Company> Companies { get; set; }
}

File: Jobby.UI/types/requests/CreateContactRequest.ts
export type CreateContactRequest = {
    "boardId": string,
    "jobIds": string[],
    "firstName": string,
    "lastName": string,
    "jobTitle": string,
    "location": string,
    "socials": Social,
    "emails": Email[],
    "phones": Phone[],
    "companies": Company[]
}

export interface Base {
    "id": string,
    "value": string,
    "type"?: number
}

type Company = Base & {
    "name": string
}

type Email = Base & {
    "name": string,
}

type Phone = Base & {
    "number": string,
}

type Social = {
    "twitterUrl": string,
    "facebookUrl": string,
    "linkedInUrl": string,
    "githubUrl": string
}


